servicetalk-http-utils/src/main/java/io/servicetalk/http/utils/AbstractTimeoutHttpFilter.java

```diff
@@ -117,7 +122,9 @@ final Single<StreamingHttpResponse> withTimeout(final StreamingHttpRequest reque
             final Duration timeout = timeoutForRequest.apply(request, useForTimeout);
             Single<StreamingHttpResponse> response = responseFunction.apply(request);
             if (null != timeout) {
-                final Single<StreamingHttpResponse> timeoutResponse = response.timeout(timeout, useForTimeout);
+                final Single<StreamingHttpResponse> timeoutResponse = response
+                        .<StreamingHttpResponse>liftSync(CleanupSubscriber::new)
```

Comment:
I feels like we need something similar inside `JavaNetSoTimeoutHttpConnectionFilter` as well. In that case, if `timeout` delivers `onError`, amb operator will do the cancel for the response and then we are in the same situation of response racing with cancel.

Reply:
How about in a followup?

Reply:
Works for me